java/vector/src/main/codegen/templates/ComplexCopier.java

```diff
@@ -63,6 +64,10 @@ private static void writeValue(FieldReader reader, FieldWriter writer) {
           writer.start();
           for(String name : reader){
             FieldReader childReader = reader.reader(name);
+            if (writer instanceof UnionMapWriter) {
```

Comment:
Why not add fill in the case for `MAP` above instead of all of this?

Reply:
Good suggestion, fixed.